frame/proxy/src/lib.rs

```diff
@@ -800,4 +798,11 @@ impl<T: Config> Pallet<T> {
 		let e = call.dispatch(origin);
 		Self::deposit_event(Event::ProxyExecuted { result: e.map(|_| ()).map_err(|e| e.error) });
 	}
+
+	pub fn remove_all_proxy_delegates(who: &T::AccountId)-> DispatchResult {
+		let mut proxies = Proxies::<T>::get();
+		let old_deposit = proxies.remove(&who);
```

Comment:
Why couldn't you use `Proxies::<T>::take(&who)` just like how it was?

Reply:
Fixed. all test passed . 
Thanks for correcting my mistakes

Reply:
@ggwpez pending for review.

Reply:
@ggwpez Hi this PR is pending for your review , can you please review it.